cmake/strict/strict_flags_helpers.cmake

```diff
@@ -65,7 +65,7 @@ function(get_gnu_base_flags flags)
     if(${cmake_compiler_version} VERSION_GREATER_EQUAL 6)
         list(APPEND gnu_flags "-Wduplicated-cond")
         list(APPEND gnu_flags "-Wlogical-op")
-        list(APPEND gnu_flags "-Wunused-const-variable=2")
+        list(APPEND gnu_flags "-Wunused-const-variable=1")
```

Comment:
This change is needed to allow the "trick" used by `WT_CONF_ID_STRUCTURE` in conf.h.  Set as `1` allows `static const` variables that are not used to be allowed in include files, but you get a warning/error if it in the compilation unit itself.  Set as `2` gives warnings for both.  I think `1` is reasonable, as it would allow us to replace some #define constants with typed static constants.

This change could be split out if it warrants more discussion.